src/Symfony/Component/ValueExporter/Exporter/ValueToStringExporter.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ValueExporter\Exporter;
+
+use Symfony\Component\ValueExporter\Formatter\StringFormatterInterface;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ * @author Quentin Schuler <qschuler@neosyne.com>
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+class ValueToStringExporter extends AbstractValueExporter
+{
+    protected $formatterInterface = StringFormatterInterface::class;
+
+    public function exportValue($value, $depth = 1, $expand = false)
+    {
+        // Arrays have to be handled first to deal with nested level and depth,
+        // this implementation intentionally ignores \Traversable values.
+        // Therefor, \Traversable instances might be treated as objects unless
+        // implementing a {@link StringFormatterInterface} and passing it to
+        // the exporter in order to support them.
+        if (is_array($value) && !is_callable($value)) {
+            if (empty($value)) {
+                return 'array()';
+            }
+            $indent = str_repeat('  ', $depth);
+
+            $a = array();
+            foreach ($value as $k => $v) {
+                if (is_array($v) && !empty($v)) {
+                    $expand = true;
+                }
+                $a[] = sprintf('%s => %s', is_string($k) ? sprintf("'%s'", $k) : $k, $this->exportValue($v, $depth + 1, $expand));
+            }
+            if ($expand) {
+                return sprintf("array(\n%s%s\n%s)", $indent, implode(sprintf(", \n%s", $indent), $a), str_repeat('  ', $depth - 1));
+            }
+
+            $s = sprintf('array(%s)', implode(', ', $a));
+
+            if (80 > strlen($s)) {
+                return $s;
+            }
+
+            return sprintf("array(\n%s%s\n)", $indent, implode(sprintf(",\n%s", $indent), $a));
+        }
+        // Not an array, test each formatter
+        foreach ($this->formatters as $formatter) {
```

Comment:
@stof So here I should use a getter ?

I've pushed some minor changes just now on that class, differences can be seen in the tests.

In fact, I've first tried to use the `VarDumper` component, but the `ValueExporter` looks more like `var_export` to me, not `var_dump`. So I took example on the architecture of the var dumper wrapper and kept the light weight of the original `ValueExporter` class, since it does not need cloners, we just want the type here not a deep profile.
I guess if it should replace the logic in many place in the core among the DataCollector profile, this method will be called many time each request. `dump` has a more specific usage imo.
So I get I'm gonna use that component as is to handle assertion messages (logs and exceptions) in the bundle I'm working on but I can close that PR if you think it's off topic.

Does this make sense to you ?

 In any case I really appreciate your review. Thanks.